tests/frontend/onnx/test_context.py

```diff
@@ -0,0 +1,189 @@
+        TensorProto,
+        ValueInfoProto,
+        helper,
+    )
+
+    from xdsl.frontend.onnx.context import (
+        OnnxXdslMapping,
+        build_module,
+        visit_graph,
+        visit_node,
+        visit_value_info,
+    )
+except ImportError as exc:
+    print(exc)
+    pytest.skip("onnx is an optional dependency", allow_module_level=True)
+
+
+def test_visit_node_unknown_op_name():
+    """
+    Test for unknown expected onnx op
+    """
+    node_attributes = {
+        "name": "dummy_name",
+        "op_type": "dummy_op",
+    }
+
+    node = helper.make_node(
+        **node_attributes, inputs=["input1", "input2"], outputs=["output"]
+    )
+    ctx = OnnxXdslMapping()
+    with pytest.raises(ValueError, match="Unknown ONNX op name dummy_op"):
+        visit_node(node=node, ctx=ctx)
+
+
+def test_visit_node_add():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Add operation
+    graph, add_node = _create_graph_binary_op("Add", "add_graph")
+
+    # visit graph
+    visit_graph(graph, ctx)
+
+    # visit node (test passes if no exceptions are raised)
+    visit_node(add_node, ctx)
+
+
+def test_visit_node_sub():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Sub operation
+    graph, sub_node = _create_graph_binary_op("Sub", "sub_graph")
+
+    # visit graph
+    visit_graph(graph, ctx)
+
+    # visit node (test passes if no exceptions are raised)
+    visit_node(sub_node, ctx)
+
+
+def test_visit_graph_add():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Add operation
+    graph, _ = _create_graph_binary_op("Add", "add_graph")
+
+    # visit graph
+    visit_graph(graph, ctx)
+
+    # check value_by_name keys
+    keys = list(ctx.value_by_name.keys())
+    assert keys == ["input1", "input2", "output"]
+
+    # check expected generated ir
+    gen_ir = ctx.value_by_name[keys[2]].owner
+    assert (
+        str(gen_ir)
+        == "%0 = onnx.Add(%1, %2) : (tensor<0x0xf32>, tensor<0x0xf32>) -> tensor<0x0xf32>"
+    )
+
+
+def test_visit_graph_sub():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Sub operation
+    graph, _ = _create_graph_binary_op("Sub", "sub_graph")
+
+    # run visit graph
+    visit_graph(graph, ctx)
+
+    # check value_by_names keys
+    keys = list(ctx.value_by_name.keys())
+    assert keys == ["input1", "input2", "output"]
+
+    # check generated ir
+    gen_ir = ctx.value_by_name[keys[2]].owner
+    assert (
+        str(gen_ir)
+        == "%0 = onnx.Sub(%1, %2) : (tensor<0x0xf32>, tensor<0x0xf32>) -> tensor<0x0xf32>"
+    )
+
+
+def test_build_module():
+    # create graph composed only of one Add operation
+    graph, _ = _create_graph_binary_op("Add", "add_graph")
+
+    # create module
+    module = build_module(graph)
+
+    # define expected output
+    expected = """builtin.module {
```

Comment:
nit:
```suggestion
    expected = """\
builtin.module {
```
It would be nice to have the `builtin.module` be on a new line, do you think you could try this change? I may be remembering the syntax incorrectly, maybe it can be done better by adding the white space we might want at the beginning and end and adding a `.strip()` to remove the white space at the end.

Reply:
Done